consult.el

```diff
@@ -1102,46 +1102,81 @@ for which the command history is used."
    ;; If pressing "C-x M-:", i.e., `repeat-complex-command',
    ;; we are instead querying the `command-history' and get a full s-expression.
    ((eq last-command 'repeat-complex-command)
-    (delete-dups (mapcar #'prin1-to-string command-history)))
+    (cons (delete-dups (mapcar #'prin1-to-string command-history))
+          'minibuffer-history-position))
    ;; In the minibuffer we use the current minibuffer history,
    ;; which can be configured by setting `minibuffer-history-variable'.
    ((minibufferp)
-    (if (fboundp 'minibuffer-history-value)
-        (minibuffer-history-value) ;; Emacs 27 only
-      (symbol-value minibuffer-history-variable)))
+    (cons (if (fboundp 'minibuffer-history-value)
+              (minibuffer-history-value) ;; Emacs 27 only
+            (symbol-value minibuffer-history-variable))
+          'minibuffer-history-position))
    ;; Otherwise we use a mode-specific history, see `consult-mode-histories'.
-   (t (when-let (history
-                 (or (seq-find (lambda (ring)
-                                 (and (derived-mode-p (car ring))
-                                      (boundp (cdr ring))))
-                               consult-mode-histories)
-                     (user-error
-                      "No history configured for `%s', see `consult-mode-histories'"
-                      major-mode)))
-        (symbol-value (cdr history))))))
+   (t (cl-destructuring-bind (history index-var)
```

Comment:
I would like to avoid cl-destructuring-bind, I don't use it anywhere. I prefer pcase.